core/data/src/main/java/uk/gov/gchq/gaffer/data/element/ReservedPropertyNames.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.data.element;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Enumeration of all protected property names.
+ */
+public enum ReservedPropertyNames {
+
+    VERTEX("vertex"),
+    SOURCE("src"),
+    DESTINATION("dst"),
```

Comment:
I'm not sure about putting this class in the data module and making it part of core Gaffer. Those names are only reserved when using spark. However, it would be useful for users to be aware of these reserved names when they first create a Gaffer schema even before they thing about using spark.

I would be happy to keep this class here if we included things like: "vertex", "VERTEX", "src", "SRC", "source", "SOURCE", etc. Otherwise we will soon end up having another library that needs to use a slightly different keywords. However, if we are saying these property names are reserved then we really should validate schemas and tell users not to use these names at all. We can't really fail schema validation as we would break any existing systems using these names. Perhaps we include some validation in the core schema validation for these property names, but just log warnings if any of these names are used. In version 2.0.0 we could change the warning to be an error.

Reply:
Done.